ateam_ui/src/src/overlay.ts

```diff
@@ -56,195 +55,209 @@ export class Overlay {
+                graphic.beginFill(overlay.fill_color);
+                graphic.lineStyle(overlay.stroke_width, overlay.stroke_color);
+                for (const point of overlay.points) {
+                    graphic.lineTo(scale * point.x, -scale * point.y);
                 }
-                break;
-            case OverlayType.Text:
-                // TEXT IS WEIRD IN PIXI
-                // The text is rendered as a new object and added
-                // as a child to the graphics object
-
-                const text = new PIXI.Text(this.text, {
-                    fontSize: this.stroke_width,
-                    fill: this.fill_color
-                });
-                text.name = "text"
-
-                text.anchor.set(0.5, 0.5);
-                text.rotation = -renderConfig.angle; // offset the rotation of the canvas so the text always appears right side up
-
-                {
-                    let graphicChild = graphic.getChildByName("text");
-
-                    if (graphicChild) {
-                        graphicChild = text;
-                    } else {
-                        graphic.addChild(text);
-                    }
+                graphic.endFill();
+            }
+            break;
+        case OverlayType.Text:
+            // TEXT IS WEIRD IN PIXI
+            // The text is rendered as a new object and added
+            // as a child to the graphics object
+
+            const text = new PIXI.Text(overlay.text, {
+                fontSize: overlay.stroke_width,
+                fill: overlay.fill_color
+            });
+            text.name = "text"
+
+            text.anchor.set(0.5, 0.5);
+            text.rotation = -renderConfig.angle; // offset the rotation of the canvas so the text always appears right side up
+
+            {
+                let graphicChild = graphic.getChildByName("text");
+
+                if (graphicChild) {
+                    graphicChild = text;
+                } else {
+                    graphic.addChild(text);
                 }
+            }
 
-                break;
-            case OverlayType.Heatmap:
-                {
-                    let graphicChild = graphic.getChildByName(this.id) as PIXI.Sprite;
-                    if (!graphicChild) {
-                        graphicChild = new PIXI.Sprite();
-                        graphicChild.name = this.id;
-                        graphic.addChild(graphicChild);
-                    }
-
-                    graphicChild.texture = this.heatmap_texture;
-                    graphicChild.width = scale * this.scale.x;
-                    graphicChild.height = scale * this.scale.y;
-                    graphicChild.x = -scale * this.scale.x / 2;
-                    graphicChild.y = -scale * this.scale.y / 2;
+            break;
+        case OverlayType.Heatmap:
+            {
+                let graphicChild = graphic.getChildByName("heatmap") as PIXI.Sprite;
```

Comment:
Does this mean we can only have one heatmap in the overlays?
This sounds vaguely familiar, so we might have talked about this in person at some point.

Reply:
the child is a member of the specific overlays graphics object so you can have one heatmap per overlay